### ISSUE TITLE
Inconsistent rounding of the `repr` of `length`, `relative`, `fraction` and `ratio` values

### ISSUE THREAD

**Issue by dixslyf** [typst/typst#2836]
### Description

Typst (tested on 0.9.0) seems to incorrectly round the `repr` of `length`, `relative`, `fraction` and `ratio` values between `1` and `1.035`.

```typ
#repr(1.005pt + 1.005em) \
#repr(1.005%) \
#repr(1.005% + 1.005pt + 1.005em) \
#repr(1.005fr) \

#repr(1.015pt + 1.015em) \
#repr(1.015%) \
#repr(1.015% + 1.015pt + 1.015em) \
#repr(1.015fr) \

#repr(1.025pt + 1.025em) \
#repr(1.025%) \
#repr(1.025% + 1.025pt + 1.025em) \
#repr(1.025fr) \

#repr(1.035pt + 1.035em) \
#repr(1.035%) \
#repr(1.035% + 1.035pt + 1.035em) \
#repr(1.035fr) \

#repr(1.045pt + 1.045em) \
#repr(1.045%) \
#repr(1.045% + 1.045pt + 1.045em) \
#repr(1.045fr) \
```

![image](https://github.com/typst/typst/assets/56017218/e64dc249-e97b-4078-a956-ccb790bcdd50)

`1.036` rounds correctly:

```typ
#repr(1.036pt + 1.036em) \
#repr(1.036%) \
#repr(1.036% + 1.036pt + 1.036em) \
#repr(1.036fr) \
```

![image](https://github.com/typst/typst/assets/56017218/dfee9506-ba8e-4f31-a036-1b2ce97df6aa)

Compare that to the following, which simply changes the first digit from `1` to `2` for each component:

```typ
#repr(2.005pt + 2.005em) \
#repr(2.005%) \
#repr(2.005% + 2.005pt + 2.005em) \
#repr(2.005fr) \

#repr(2.015pt + 2.015em) \
#repr(2.015%) \
#repr(2.015% + 2.015pt + 2.015em) \
#repr(2.015fr) \

#repr(2.025pt + 2.025em) \
#repr(2.025%) \
#repr(2.025% + 2.025pt + 2.025em) \
#repr(2.025fr) \

#repr(2.035pt + 2.035em) \
#repr(2.035%) \
#repr(2.035% + 2.035pt + 2.035em) \
#repr(2.035fr) \

#repr(2.045pt + 2.045em) \
#repr(2.045%) \
#repr(2.045% + 2.045pt + 2.045em) \
#repr(2.045fr) \
```

![image](https://github.com/typst/typst/assets/56017218/49bf87a9-d99e-4a1f-9bde-6ce12cd76f8b)

As evident, `em` components are also not rounded.

I would argue that none of the components should be rounded, since `repr` (as stated in the documentation) is for debugging purposes. There's not much use for `repr` for debugging if it doesn't precisely represent the actual value. With the fields added by https://github.com/typst/typst/pull/790, you can, of course, work around this, but I think my point still stands.

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
"I would argue that none of the components should be rounded"

Some rounding is always necessary, since most of these aren't exactly representable as floats

**Comment by dixslyf** [comment]
> Some rounding is always necessary, since most of these aren't exactly representable as floats

Yes, of course, since floats have fixed precision. What I meant was I'd expect `repr` to use the underlying float value as is rather than round it to 2 decimal places ([source for Abs](https://github.com/typst/typst/blob/9926a594e7f462103b47930270a00c1b9ce3cbf3/crates/typst/src/layout/abs.rs#L151), for example).
